netmanager/src/views/pages/Map/OverlayMap.js

```diff
@@ -143,47 +134,54 @@ const PollutantSelector = ({ className, onChange, showHeatMap }) => {
       <span>
         NO<sub>2</sub>
       </span>
-    ),
+    )
   };
 
   const onHandleClick = () => {
     setOpen(!open);
   };
 
   const handleMenuItemChange = (pollutant, state) => () => {
-    setPollutant(pollutant);
+    localStorage.pollutant = pollutant;
     setOpen(!open);
     onChange(state);
+    window.location.reload();
```

Comment:
Reloads the entire window once the user clicks to change pollutant value

Reply:
Thanks @Codebmk for the changes. This is amazing and much welcome!

This is amazing!

I am having some reservations with the reload of the entire page whenever a user changes the concentration levels. I find it to be a somehow bad user experience.

Could we explore the possibilities that do not require the refresh of the entire page/map?